src/index.ts

```diff
@@ -5,18 +5,21 @@ export interface SimpleChange<T> {
     isFirstChange: () => boolean;
 }
 
-export function OnChange<T = any>(callback: (value: T, simpleChange?: SimpleChange<T>) => void) {
+export function OnChange<T = any>(callback: ((value: T, simpleChange?: SimpleChange<T>) => void) | string) {
     const cachedValueKey = Symbol();
     const isFirstChangeKey = Symbol();
+
+    const isFunction = (fn: any) => fn.constructor === Function;
+
     return (target: any, key: PropertyKey) => {
         Object.defineProperty(target, key, {
             set: function (value) {
+
+                const callBackFn = (isFunction(callback) ? callback : target[callback as string]) as Function;
```

Comment:
Please throw an error when the function doesn't exist.

Reply:
Added the exception handling when developer passes invalid string to OnChange decorator